src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,134 @@
+                xRandom = rand.nextInt(maxValue + 1);
+                yRandom = rand.nextInt(maxValue + 1);
+
+                Vector2d newRandomVector = new Vector2d(xRandom, yRandom);
+
+                for (Grass field : this.grasses) {
+                    if (field.getPosition().equals(newRandomVector)) {
+                        repeat = true;
+                    }
+                }
+
+                if (!repeat) {
+                    this.grasses.add(new Grass(newRandomVector));
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        return ! super.isOccupied(position);
+    }
+
+    @Override
+    public boolean place(Animal animal) {
+        return super.place(animal);
+    }
+
+    @Override
+    public boolean isOccupied(Vector2d position) {
+        boolean bool = super.isOccupied(position);
+
+        if (bool){
+            return true;
+        }else {
+            for (Grass grass: this.grasses){
+                if (grass.getPosition().equals(position)){
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+
+    @Override
+    public Object objectAt(Vector2d position) {
+        Object object = super.objectAt(position);
+        if (object != null){
+            return object;
+        }
+        else {
+            for (Grass grass: this.grasses){
+                if (grass.getPosition().equals(position)){
+                    return grass;
+                }
+            }
+            return null;
+        }
+    }
+    @Override
+    public String toString() {
+
+        int minX = this.grasses.get(0).getPosition().x;
+        int maxX = this.grasses.get(0).getPosition().x;
+        int minY = this.grasses.get(0).getPosition().y;
+        int maxY = this.grasses.get(0).getPosition().y;
+        int currentX;
+        int currentY;
+        for(Animal animal: animals){
+            currentX = animal.getPosition().x;
+            currentY = animal.getPosition().y;
+            if (minX > currentX) {
+                minX = currentX;
+            }
+            if (minY > currentY){
+                minY = currentY;
+            }
+            if (maxX < currentX){
+                maxX = currentX;
+            }
+            if (maxY < currentY){
+                maxY = currentY;
+            }
+        }
+        for(Grass grass: this.grasses){
+            currentX = grass.getPosition().x;
+            currentY = grass.getPosition().y;
+            if (minX > currentX) {
+                minX = currentX;
+            }
+            if (minY > currentY){
+                minY = currentY;
+            }
+            if (maxX < currentX){
+                maxX = currentX;
+            }
+            if (maxY < currentY){
+                maxY = currentY;
+            }
+        }
+        return super.toString(new Vector2d(minX,minY), new Vector2d(maxX, maxY));
```

Comment:
to działa, ale takie rozwiązanie nie wykorzystuje potencjału klas abstrakcyjnych (moglibyśmy równie dobrze umieścić takiego toStringa z dwoma argumentami w jakiejś dedykowanej klasie utilowej). Lepiej byłoby to zrobić w drugą stronę - niech klasa abstrakcyjna woła metody abstrakcyjne do liczenia pozycji, które byłyby zaklepane w klasach dziedziczących.